llama-index-utils/llama-index-utils-workflow/llama_index/utils/workflow/draw.py

```diff
@@ -53,6 +53,20 @@ def draw_all_possible_flows(
                 shape="ellipse",
             )  # Light green for events
 
+        for return_type in step_config.return_types:
+            print(f"{return_type}", flush=True)
+            if return_type == type(None):
+                continue
+
+            if issubclass(return_type, BlockingEvent):
```

Comment:
I wonder if we should insert some generic "external event" into the diagram to connect the blocking and unblocking events

Reply:
I think it would be an external step

So like, any blocking event would be linked to some (grey?) step labelled "External Step", and that step emits any unblocking event

Just thought it might make the generated diagram look nicer

Reply:
![image](https://github.com/user-attachments/assets/96cbad0d-1455-4d87-bd5d-b967acf1e2ff)

I was in the process of doing this and chose "grey-ish" color for it haha. Happy to make changes to the style if we're still not happy with it.

Reply:
Yea that looks pretty good!